hazelcast-wm/src/main/java/com/hazelcast/web/ClusteredSessionService.java

```diff
@@ -104,18 +103,14 @@ public ClusteredSessionService(FilterConfig filterConfig, Properties properties,
 
         es.scheduleWithFixedDelay(new Runnable() {
             public void run() {
-                while (ensureInstanceRunning) {
-                    try {
-                        ensureInstance();
-                    } catch (Exception ignored) {
-                        if (ignored instanceof InterruptedException) {
-                            ensureInstanceRunning = false;
-                        }
-                        EmptyStatement.ignore(ignored);
-                    }
+                try {
+                    ensureInstance();
+                } catch (Exception ignored) {
+                    EmptyStatement.ignore(ignored);
                 }
             }
         }, 0, CLUSTER_CHECK_INTERVAL, TimeUnit.SECONDS);
```

Comment:
@jnehlmeier Thanks, for your suggestion. You are right, making synchronous call first is better approach.  I accidentally closed this PR, you can see new PR here https://github.com/hazelcast/hazelcast/pull/5832.